buildpack/v06/encoderdecoder.go

```diff
@@ -0,0 +1,56 @@
+package v06
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/BurntSushi/toml"
+
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/buildpack/layertypes"
+)
+
+type typesTable struct {
+	Build  bool `toml:"build"`
+	Launch bool `toml:"launch"`
+	Cache  bool `toml:"cache"`
+}
+
+type layerMetadataTomlFile struct {
+	Data  interface{} `toml:"metadata"`
+	Types typesTable  `toml:"types"`
+}
+
+type EncoderDecoder06 struct {
```

Comment:
Can this struct be private? `encoderDecoder06`

Reply:
Unless I'm missing something, I don't think it's possible.
We're returning this struct in `NewEncoderDecoder` and calling to all of the functions on it from `buildpack/files.go`.

Reply:
As long as you aren't constructing structs directly via `decoder := EncoderDecoder06{}`, `NewEncoderDecoder` and the methods you want to expose is enough. The struct itself can remain private and not exposed to devs importing the package. This would restrict them to use `NewEncoderDecoder` to create a `dncoderDecoder06` as well.

Reply:
I'm sorry, my fault. I got an error last night from the linter and I was too tired to actually read it :)
Anyway, you're right. Please see https://github.com/buildpacks/lifecycle/pull/537/commits/2373122d4321c49d25e9d495e98b6755013b4095.